use-location.js

```diff
@@ -29,16 +29,24 @@ const subscribeToLocationUpdates = (callback) => {
 export const useLocationProperty = (fn, ssrFn) =>
   useSyncExternalStore(subscribeToLocationUpdates, fn, ssrFn);
 
-const currentSearch = () => location.search;
+const currentSearch = () => {
+  if (typeof location != "undefined") location.search;
+};
+
 export const useSearch = () => useLocationProperty(currentSearch);
 
-const currentPathname = () => location.pathname;
+const currentPathname = () => {
+  if (typeof location != "undefined") return location.pathname;
+};
 
-export const usePathname = ({ ssrPath } = {}) =>
-  useLocationProperty(
+export const usePathname = ({ ssrPath } = {}) => {
+  if (ssrPath) return ssrPath;
```

Comment:
This check is redundant, because React is expected to handle that in `useSyncExternalStore`. See line 47 below, when `ssrPath` is given the second callback becomes a stub that always returns this value.